language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ForeachUtil.java

```diff
@@ -0,0 +1,189 @@
+ *  under the License.
+ */
+package org.ballerinalang.langserver.completions.util;
+
+import io.ballerina.compiler.api.symbols.ArrayTypeSymbol;
+import io.ballerina.compiler.api.symbols.MapTypeSymbol;
+import io.ballerina.compiler.api.symbols.StreamTypeSymbol;
+import io.ballerina.compiler.api.symbols.TableTypeSymbol;
+import io.ballerina.compiler.api.symbols.TupleTypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeDescKind;
+import io.ballerina.compiler.api.symbols.TypeReferenceTypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.syntax.tree.BlockStatementNode;
+import io.ballerina.compiler.syntax.tree.ExpressionStatementNode;
+import io.ballerina.compiler.syntax.tree.FieldAccessExpressionNode;
+import io.ballerina.compiler.syntax.tree.FunctionBodyNode;
+import io.ballerina.compiler.syntax.tree.SimpleNameReferenceNode;
+import org.ballerinalang.langserver.common.utils.CommonUtil;
+import org.ballerinalang.langserver.commons.BallerinaCompletionContext;
+import org.ballerinalang.langserver.commons.completion.LSCompletionItem;
+import org.ballerinalang.langserver.completions.StaticCompletionItem;
+import org.ballerinalang.langserver.completions.builder.ForeachCompletionItemBuilder;
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Dynamic foreach snippet construction for iterables.
+ */
+public class ForeachUtil {
+
+    private static final String VAR_NAME = "item";
+    private static final String VAR_TYPE = "var";
+    private static final List<TypeDescKind> ITERABLES = Arrays.asList(
+            TypeDescKind.STRING, TypeDescKind.ARRAY, TypeDescKind.TUPLE,
+            TypeDescKind.MAP, TypeDescKind.RECORD, TypeDescKind.TABLE,
+            TypeDescKind.STREAM, TypeDescKind.XML);
+
+    private ForeachUtil() {
+    }
+
+    /**
+     * Given a expression node in a particular completion context,
+     * validate if the expression is an iterable and returns the corresponding foreach completion item.
+     *
+     * @param ctx        completion context
+     * @param expr       expression node
+     * @param typeSymbol type symbol corresponding to the expression node
+     * @return a list of {@link LSCompletionItem}.
+     */
+    public static List<LSCompletionItem> getForeachCompletionItemForIterable(BallerinaCompletionContext ctx,
+                                                                             FieldAccessExpressionNode expr,
+                                                                             Optional<TypeSymbol> typeSymbol) {
+        List<LSCompletionItem> completionItems = new ArrayList<>();
+        if (isInBlockContext(expr) && expr.expression() instanceof SimpleNameReferenceNode) {
```

Comment:
Let's avoid the `instanceof` check and instead use the `kind` check. Also, use the early return here.